conn.go

```diff
@@ -118,12 +118,12 @@ func (c *Conn) Streams() []*Stream {
 // Close closes this connection
 func (c *Conn) Close() error {
 	c.closeLock.Lock()
-	defer c.closeLock.Unlock()
-
 	if c.closed {
+		c.closeLock.Unlock()
```

Comment:
thats also something we don't want though, we would end up with tons of random goroutines hanging around for an indeterminate period of time.

Are you opposed to a method that calls close, and exits early if its being closed already like this one? maybe `StartClosing` or something?